src/load.ts

```diff
@@ -0,0 +1,59 @@
+import Promise from './Promise';
+
+declare var define: {
+	(...args: any[]): any;
+	amd: any;
+};
+
+export interface AMDRequire {
+	(moduleIds: string[], callback: (...modules:any[]) => void): void;
+}
+export interface NodeRequire {
+	(moduleId: string): any;
+}
+export type Require = AMDRequire | NodeRequire;
+
+export interface Load {
+	(require: Require|string, ...moduleIds: string[]): Promise<any[]>;
```

Comment:
I think you implementation logic is 100% correct.  I just think the exposed API could be a bit cleaner (though it is technically correct).  Let's just admit, the availability of union types of blinded us from thinking of overloading things!  I have found myself getting wound into knots with union types only to realises that overloading actually makes things cleaner.